python/ray/train/base_trainer.py

```diff
@@ -624,9 +621,22 @@ def _save(self, experiment_path: Union[str, Path]):
         set of parameters can be passed in again), the argument will be loaded
         from this saved one.
         """
+        param_dict = self._param_dict.copy()
+        datasets = param_dict.pop("datasets", {})
+
+        def raise_fn():
+            raise RuntimeError
```

Comment:
Should this raise an actionable message? Or is this never surfaced to the user?
Can you add some in-line comments to explain?

Reply:
I don't think this should ever surface to the user. We will throw an error if they try restoring without passing in datasets again. I just needed some way to throw a dummy error in the case that the user tries to deserialize the saved trainer themselves and use it. I can add some comments though!